src/databricks/labs/remorph/reconcile/connectors/data_source.py

```diff
@@ -10,14 +10,27 @@
 )
 
 
+class SecretsProvider:
+    def get_secret(self, ws: WorkspaceClient, scope: str, key: str):
```

Comment:
```suggestion
    def get_secret(self, scope: str, key: str):
```
Workspace client has to be a private property here. Though, looking at the body - you can just pass ws.secrets.get_secret as Callable[[str,str],str] and avoid this class

Reply:
Made the get_secret method public and removed the abstraction and mocking simply.